docs/data/charts/bars/BorderRadius.tsx

```diff
@@ -1,52 +1,92 @@
 import * as React from 'react';
-import { BarChart } from '@mui/x-charts/BarChart';
+import { BarChart, BarChartProps } from '@mui/x-charts/BarChart';
 import { axisClasses } from '@mui/x-charts/ChartsAxis';
+import Stack from '@mui/material/Stack';
+import { HighlightedCode } from '@mui/docs/HighlightedCode';
+import TextField from '@mui/material/TextField';
+import MenuItem from '@mui/material/MenuItem';
+import Slider from '@mui/material/Slider';
+import Typography from '@mui/material/Typography';
 
 export default function BorderRadius() {
+  const [layout, setLayout] = React.useState<'horizontal' | 'vertical'>('vertical');
+  const [radius, setRadius] = React.useState(10);
+
   return (
-    <div style={{ width: '100%' }}>
+    <Stack direction="column" spacing={1} sx={{ width: '100%', maxWidth: 600 }}>
+      <Stack direction="row" spacing={1}>
+        <Stack direction="column" spacing={1} flex={1}>
+          <Typography gutterBottom>Border Radius</Typography>
+          <Slider
+            value={radius}
+            onChange={(e, v) => setRadius(v as number)}
+            valueLabelDisplay="auto"
+            min={0}
+            max={50}
+            sx={{ mt: 2 }}
+          />
+        </Stack>
+        <TextField
+          select
+          sx={{ minWidth: 150 }}
+          label="layout"
+          value={layout}
+          onChange={(event) =>
+            setLayout(event.target.value as 'horizontal' | 'vertical')
+          }
+        >
+          <MenuItem value="horizontal">Horizontal</MenuItem>
```

Comment:
Super nitpicky nitpick: There is this one thing that throws me off a bit about this demo when the chart has a horizontal layout: this border radius here in the corner even though this bar is stuck to the y axis.
![Screenshot 2024-05-07 113829](https://github.com/mui/mui-x/assets/72460825/1610cfe9-afd8-4bb9-b303-983237e19009)

Now, I know that it's how it's supposed to work, so there is no problem with the functionality per se. But maybe we can adjust the values just a little bit for this demo to have a little padding between the bar and the axis? Again, very nitpicky, so again, if it's a pain, feel free to ignore 🙈

Reply:
I've updated the values so they don't touch the sides. It should work on most screens, but for consistent padding the users should probably set their own padding, or set the axis to contain values bigger than the current values.

Reply:
> users should probably set their own padding, or set the axis to contain values bigger than the current values.

makes sense👌 it's looking really good in this demo ❤️